### PR TITLE
Add tests to bump up coverage

### PR DESCRIPTION
**Pull request by scb01**
Add unit and integration tests to bump up coverage to 70%.

### CODE REVIEW

File: tests/integration_test.rs [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        .unwrap()
+        .scope_logs
+        .first()
+        .unwrap()
+        .log_records
+        .first()
+        .unwrap()
+        .body;
+    let log_message = log_message.clone().unwrap().value.unwrap();
+    assert_eq!(log_message, Value::StringValue(warn_log.to_owned()));
```
Comment (nlcamp) [inline tests/integration_test.rs]: Shouldn't the `filtered_logs` and `unfiltered_logs` channels be drained completely to make sure only the expected messages flow through? It looks like this code only examines the first message sent over each channel.

File: tests/integration_test.rs [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    Otel,
+};
+
+use tokio::sync::mpsc::Receiver;
+use tokio::time::timeout;
+
+mod mocks;
+
+#[tokio::test]
+async fn end_to_end_test() {
```
Comment (nlcamp) [inline tests/integration_test.rs:41]: Could you add a brief comment explaining what this test does?

**Review by nlcamp — APPROVED** [review]

File: tests/integration_test.rs (hunk 2 above)
Comment (scb01) [reply tests/integration_test.rs:41]: Done.

File: tests/integration_test.rs (hunk 1 above)
Comment (scb01) [reply tests/integration_test.rs]: Good call. Adjusted so that we wait for log messages to be drained.

**Pull request merged**